test/androidTest/java/org/thoughtcrime/securesms/RepliesEspressoTest.java

```diff
@@ -0,0 +1,82 @@
+package org.thoughtcrime.securesms;
+
+import android.support.test.filters.LargeTest;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.thoughtcrime.securesms.espresso.Helper;
+
+@RunWith(AndroidJUnit4.class)
+@LargeTest
+public class RepliesEspressoTest {
+    @Rule
+    public ActivityTestRule<ConversationListActivity> mainActivityRule =
+            new ActivityTestRule(ConversationListActivity.class, true, false);
+
+    @Test
+    public void buttonExists(){
+        Helper helper = new Helper(mainActivityRule);
+        String testSend = helper.randString();
+
+        helper
+                .goConversations()
+                .goConversation()
+                .sendMessage(testSend)
+                .selectMessage(0)
+                .assertId(R.id.menu_context_reply);
+
+    }
+
+    @Test
+    public void canSendReply() {
+        Helper helper = new Helper(mainActivityRule);
+        String testSend = helper.randString();
+        String testReply = helper.randString();
+
+        helper
+                .goConversations()
+                .goConversation()
+                .sendMessage(testSend)
+                .sendReply(0, testReply)
+                .assertText(testReply);
+    }
+
+    @Test
+    public void replyIsVisibleInUI() {
+        Helper helper = new Helper(mainActivityRule);
+
+        String testSend = helper.randString();
+        String testReply = helper.randString();
+
+        helper
+                .goConversations()
+                .goConversation()
+                .sendMessage(testSend)
+                .sendReply(0, testReply)
+                .assertText(testReply);
```

Comment:
this asserts that the reply is visible, but not necessarily that it is in the correct message. Have you tried using `assertTextAt`?

Reply:
Tried, had a lot of trouble with it. Settled with this , and generated random strings instead to skirt around this. Assumed if its displaying, its in the right spot. Naive, maybe.